### DISCUSSION TITLE
Need help with API for adding/Inviting users to my app via link /OR code and phone number

### DISCUSSION THREAD

**Discussion by YourSportz** [community/community#123887] [Programming Help]
### Select Topic Area

Question

### Body

Need help with API for adding/Inviting users to my app via link /OR code and phone number 

Your Guidance and support is appreciated

**Comment by Aditya020224** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> Need help with API for adding/Inviting users to my app via link /OR code and phone number
> 
> Your Guidance and support is appreciated

👇👇
Most backend-as-a-service (BaaS) platforms or frameworks will provide functionalities which will help you:

1.User Management APIs: Many BaaS platforms offer APIs for user registration and invitation management. Explore the documentation of your chosen platform for specifics.
2.Custom API Development: You can develop custom API endpoints on your server to handle invitation generation, record creation, and user account linking based on the unique identifier.

**Reply by YourSportz** [reply]
Basically i am creating a football app where in user can invite players , add players via phone ,QR code or link 
Backend is Node.js

**Reply by Aditya020224** [reply]
its just for understanding purpose u can customize them further based on your needs like  using libraries like 'qrcode' for QR code generation, exploring third-party services for SMS verification for a smoother experience etc.

1.Invite via Link:
API Endpoint: POST /invites
Body: { "email": "user@example.com" } (optional phone number for fallback)
Server generates a unique invite link and sends it to the provided email.
Clicking the link redirects user to a signup/login page with pre-filled information (optional).

2. Invite via QR Code:
API Endpoint: GET /invites/qr-code?email=user@example.com (optional phone number)
Server generates a QR code containing the invite link.
User scans the QR code with their phone camera, which opens the invite link.

3. Add via Phone Number:
API Endpoint: POST /users (with phone number in request body)
Server validates phone number format.
Sends a verification code via SMS to the phone number.
User enters the verification code on your app to complete registration.

**Reply by Aditya020224** [reply]
can you please help me by marking my answer as marked answer  
It may help me as well

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @YourSportz !

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by RafaelJohn9** [comment]
@YourSportz  if you feel your problem hasn't been solved, you can always inquire for more assistance.
